src/main/java/org/gridsuite/securityanalysis/server/SecurityAnalysisController.java

```diff
@@ -72,17 +88,18 @@ public ResponseEntity<Mono<SecurityAnalysisResult>> run(@Parameter(description =
                                                             @Parameter(description = "Provider") @RequestParam(name = "provider", required = false) String provider,
                                                             @Parameter(description = "reportUuid") @RequestParam(name = "reportUuid", required = false) UUID reportUuid,
                                                             @Parameter(description = "reporterId") @RequestParam(name = "reporterId", required = false) String reporterId,
-                                                            @RequestBody(required = false) SecurityAnalysisParameters parameters) {
-        SecurityAnalysisParameters nonNullParameters = getNonNullParameters(parameters);
+                                                            @RequestBody(required = false) SecurityAnalysisParametersInfos parameters) {
+        String providerToUse = provider != null ? provider : service.getDefaultProvider();
```

Comment:
All this code could be done in the service instead of the controller

Reply:
I just leave service.getDefaultProvider() here (because of SecurityAnalysisService access)